### PR TITLE
*: start GCWorker after bootstrap finish.

### PR DESCRIPTION
**Pull request by tiancaiamao**
GCWorker should not be start when newTikvStore, because
the system maybe not bootstrap yet. Move the logic to
BootstrapSession.

@disksing @zimulala

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -392,8 +392,10 @@ func (m *MockFailure) getValue() bool {
 	return m.val
 }
 
-type etcdBackend interface {
+// EtcdBackend is used for judging a storage is a real TiKV.
```
Comment (shenli) [inline domain/domain.go]: The comment makes me confused.

Comment (tiancaiamao) [inline domain/domain.go]: Well, `kv.Storage` is an common interface, we'd better keep it clean.
But some additional work is necessary for specific storage engine, so it is defined.
We use it like this:

```
if rawStore, ok := store.(EtcdBackend); ok {
    // do something...
}
```

**Comment by zimulala** [comment]
Do you fix this issue #3449 ?

**Comment by tiancaiamao** [comment]
Exactly. @zimulala

**Comment by tiancaiamao** [comment]
PTAL @zimulala

File: store/tikv/gc_worker.go [hunk 2]
```diff
@@ -67,7 +67,10 @@ func NewGCWorker(store kv.Storage) (*GCWorker, error) {
 	}
 	var ctx goctx.Context
 	ctx, worker.cancel = util.WithCancel(goctx.Background())
-	go worker.start(ctx)
+	var wg sync.WaitGroup
```
Comment (zimulala) [inline store/tikv/gc_worker.go]: Add comments to wg's use.

**Comment by zimulala** [comment]
Rest LGTM

File: store/tikv/gc_worker.go (hunk 2 above)
Comment (tiancaiamao) [inline store/tikv/gc_worker.go]: Done.

**Comment by zimulala** [comment]
LGTM

File: domain/domain.go (hunk 1 above)
Comment (disksing) [inline domain/domain.go]: Whether the storage can run GC has nothing to with whether it uses etcd. So, I think it's better to use 2 interfaces, base on the 'minimum interface principle'.

File: store/tikv/gc_worker.go (hunk 2 above)
Comment (disksing) [inline store/tikv/gc_worker.go]: Normally, `wg.Done()` is called when a goroutine exits. Seems to use a channel instead is better.

**Pull request merged**